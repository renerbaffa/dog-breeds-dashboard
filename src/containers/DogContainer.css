.container {
  position: absolute;
  height: 100%;
  width: 100%;
}

.title {
  display: flex;
  justify-content: center;
  font-weight: bold;
  padding: 8px;
  font-size: 28px;
  padding-top: 60px;
}

.imageContainer {
  height: calc(100% - 205px);
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
}

.image {
  max-width: 600px;
  max-height: auto;
}

.buttonContainer {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 8px;
  margin: 12px 0px;
  height: 60px;
}

.button {
  padding: 12px;
}

.buttonIcon {
  margin-right: 8px;
}
