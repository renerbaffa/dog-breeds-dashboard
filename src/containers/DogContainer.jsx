import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import PlusButton from 'react-icons/lib/fa/plus';

import Button from '../components/shared/Button';
import Image from '../components/shared/Image';

import styles from './DogContainer.css';

const DogContainer = ({ className, title, ...other }) => (
  <div
    {...other}
    className={cx(className, styles.container)}
  >
    <div className={styles.title}>{title}</div>
    <div className={styles.imageContainer}>
      <Image
        className={styles.image}
        src="https://dog.ceo/api/img/shihtzu/n02086240_8108aaaa.jpg"
      />
    </div>
    <div className={styles.buttonContainer}>
      <Button className={styles.button}>
        <PlusButton className={styles.buttonIcon} />
        Load other image
      </Button>
    </div>
  </div>
);

DogContainer.propTypes = {
  className: PropTypes.string,
  title: PropTypes.string,
};

DogContainer.defaultProps = {
  className: '',
  title: 'Please select a breed',
};

export default DogContainer;
