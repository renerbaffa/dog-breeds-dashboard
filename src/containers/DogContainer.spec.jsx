import React from 'react';
import renderer from 'react-test-renderer';

import DogContainer from './DogContainer';

describe('<DogContainer />', () => {
  it('should render correctly', () => {
    const container = renderer.create(<DogContainer />).toJSON();
    expect(container).toMatchSnapshot();
  });
});
