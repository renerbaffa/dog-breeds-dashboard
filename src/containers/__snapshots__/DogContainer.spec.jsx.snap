// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DogContainer /> should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="title"
  >
    Please select a breed
  </div>
  <div
    className="imageContainer"
  >
    <div
      className="container loader"
    >
      <div>
        <svg
          height={80}
          viewBox="0 0 38 38"
          width={80}
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <linearGradient
              id="a"
              x1="8.042%"
              x2="65.682%"
              y1="0%"
              y2="23.865%"
            >
              <stop
                offset="0%"
                stopColor="gray"
                stopOpacity="0"
              />
              <stop
                offset="63.146%"
                stopColor="gray"
                stopOpacity=".631"
              />
              <stop
                offset="100%"
                stopColor="gray"
              />
            </linearGradient>
          </defs>
          <g
            fill="none"
            fillRule="evenodd"
          >
            <g
              transform="translate(1 1)"
            >
              <path
                d="M36 18c0-9.94-8.06-18-18-18"
                id="Oval-2"
                stroke="url(#a)"
                strokeWidth="2"
              >
                <animateTransform
                  attributeName="transform"
                  dur="0.9s"
                  from="0 18 18"
                  repeatCount="indefinite"
                  to="360 18 18"
                  type="rotate"
                />
              </path>
              <circle
                cx="36"
                cy="18"
                fill="#fff"
                r="1"
              >
                <animateTransform
                  attributeName="transform"
                  dur="0.9s"
                  from="0 18 18"
                  repeatCount="indefinite"
                  to="360 18 18"
                  type="rotate"
                />
              </circle>
            </g>
          </g>
        </svg>
      </div>
    </div>
  </div>
  <div
    className="buttonContainer"
  >
    <button
      className="button main"
    >
      <svg
        className="buttonIcon"
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m35.9 16.4v4.3q0 0.9-0.6 1.5t-1.5 0.7h-9.3v9.2q0 0.9-0.6 1.6t-1.5 0.6h-4.3q-0.9 0-1.5-0.6t-0.7-1.6v-9.2h-9.3q-0.9 0-1.5-0.7t-0.6-1.5v-4.3q0-0.9 0.6-1.5t1.5-0.6h9.3v-9.3q0-0.9 0.7-1.5t1.5-0.6h4.3q0.9 0 1.5 0.6t0.6 1.5v9.3h9.3q0.9 0 1.5 0.6t0.6 1.5z"
          />
        </g>
      </svg>
      Load other image
    </button>
  </div>
</div>
`;
